bundles/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/DSMRBinding.java

```diff
@@ -0,0 +1,237 @@
+	private DSMRPort dsmrPort;
+
+	/*
+	 * the refresh interval which is used to poll values from the DSMR server
+	 * 
+	 * Since we the device only updates every 10 seconds we let OpenHab
+	 * introduce a pause before execute is called again.
+	 * 
+	 * We use here half the update interval time so we have some time
+	 * to read the serial port (refreshInterval starts after previous executes ends) 
+	 */
+	private long refreshInterval = DSMR_UPDATE_INTERVAL / 2;
+
+	/**
+	 * Default Constructor
+	 */
+	public DSMRBinding() {
+	}
+
+	/**
+	 * Activate the binding. We don't do anything special here.
+	 * <p>
+	 * To simplify synchronization issues we initiate the DSMR Port
+	 * in the execute() method
+	 */
+	public void activate() {
+		logger.debug("Activate DSMRBinding");
+	}
+
+	/**
+	 * Deactivates the binding.
+	 * <p>
+	 * This will close the DSMR Port
+	 */
+	public void deactivate() {
+		// deallocate resources here that are no longer needed and
+		// should be reset when activating this binding again
+		logger.info("Deactivating DSMRBinding");
+		if (dsmrPort != null) {
+			logger.debug("Closing DSMR port");
+			dsmrPort.close();
+		} else {
+			logger.info("DSMR port was not initialised");
+		}
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected long getRefreshInterval() {
+		return refreshInterval;
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected String getName() {
+		return "DSMR Binding";
+	}
+
+	/**
+	 * The execute method of the DSMR binding will execute if:
+	 * <p>
+	 * <ul>
+	 * <li>Global binding is properly configured (i.e. port and version are set properly)
+	 * <li>Item bindings are present
+	 * </ul>
+	 * <p>
+	 * If there is no DSMR port yet, or the port is closed for some reason a new DSMR Port
+	 * is created.
+	 * <p>
+	 * The DSMR port is read and received messages are posted on the bus.
+	 */
+	@Override
+	protected void execute() {
+		// the frequently executed code (polling) goes here ...
+		logger.debug("execute() method is called!");
+
+		// Check valid global Binding configuration
+		if (!isProperlyConfigured()) {
+			logger.debug("Binding is not yet configured");
+			return;
+		}
+
+		// Check if there are any item bindings
+		if (!bindingsExist()) {
+			logger.debug("There is no existing DSMR binding configuration => refresh cycle aborted!");
+			return;
+		}
+		
+		// Check if a valid DSMR port exists. Open a new one if necessary
+		if (dsmrPort == null || !dsmrPort.isOpen()) {
+			logger.debug("Creating DSMR Port:" + port);
+			dsmrPort = new DSMRPort(port, version, DSMR_UPDATE_INTERVAL);
+		}
+
+		// Read the DSMRPort
+		List<OBISMessage<? extends State>> messages = dsmrPort.read();
+
+		// Publish messages on the event bus
+		for (OBISMessage<? extends State> msg : messages) {
+			logger.debug("Read message:" + msg);
+			for (DSMRBindingProvider provider : providers) {
+				for (String itemName : provider.getItemNames()) {
+					String item = provider.getItem(itemName);
```

Comment:
i am not sure what is meant exactly with "item" here? Since "Item" is a reserved word in openHAB please use a more meaningful variable name here